lib/org/chromium/webidl/Canonicalizer.js

```diff
@@ -0,0 +1,139 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'Canonicalizer',
+
+  documentation: `This component expects to receive AST nodes. Canonical versions
+    of definitions will be created from AST nodes.
+
+    Please note that canonicalization for same IDL fragment repository cannot
+    be distributed. All computations for the same repository must be done
+    through one canonicalizer instances, or you will get fragmented results.`,
+
+  requires: [
+    'org.chromium.webidl.ast.Enum',
+    'org.chromium.webidl.ast.Typedef',
+    'org.chromium.webidl.PipelineMessage',
+  ],
+
+  imports: [ 'error' ],
+
+  properties: [
+    // TODO: This component should not be responsible for observing the
+    // pipeline. onDone and waitTime should be removed in the future and
+    // replaced with a subscription to a component that is responsible for
+    // observing the pipeline.
+    {
+      class: 'Function',
+      documentation: `onDone is called after waitTime seconds has passed since
+        the last call to addFragment. We currently have no way of determining
+        when the Canonicalizer is finished receiving fragments. It is assumed
+        that if waitTime has passed and no new fragments are received, then
+        it will not receive any more fragments and results can be passed on.`,
+      name: 'onDone',
+      required: true,
+    },
+    {
+      class: 'Int',
+      documentation: `The amount of time to wait in seconds after the last
+        addFragment request, before onDone() is called with the canonical
+        IDL files.`,
+      name: 'waitTime',
+    },
+    {
+      documentation: `A timer that is used to determine whether waitTime
```

Comment:
In Node, setTimeout doesn't return an ID, it returns a `Timeout` object https://nodejs.org/dist/latest-v6.x/docs/api/timers.html